Tests/test_GenBank.py

```diff
@@ -3192,92 +3194,84 @@ class GenBankTests(unittest.TestCase):
-        def parse_invalid_product_line():
-            rec = SeqIO.read(os.path.join('GenBank', 'invalid_product.gb'),
-                             'genbank')
-        self.assertRaises(ValueError, parse_invalid_product_line)
+        path = os.path.join('GenBank', 'invalid_product.gb')
+        self.assertRaises(ValueError, SeqIO.read, path, 'genbank')
 
     def test_genbank_read(self):
         """GenBank.read(...) simple test."""
-        with open(os.path.join("GenBank", "NC_000932.gb")) as handle:
+        path = os.path.join("GenBank", "NC_000932.gb")
+        with open(path) as handle:
             record = GenBank.read(handle)
         self.assertEqual(['NC_000932'], record.accession)
 
     def test_genbank_read_multirecord(self):
         """GenBank.read(...) error on multiple record input."""
-        with open(os.path.join("GenBank", "cor6_6.gb")) as handle:
+        path = os.path.join("GenBank", "cor6_6.gb")
+        with open(path) as handle:
             self.assertRaises(ValueError, GenBank.read, handle)
 
     def test_genbank_read_invalid(self):
         """GenBank.read(...) error on invalid file (e.g. FASTA file)."""
-        with open(os.path.join("GenBank", "NC_000932.faa")) as handle:
+        path = os.path.join("GenBank", "NC_000932.faa")
+        with open(path) as handle:
             self.assertRaises(ValueError, GenBank.read, handle)
 
     def test_genbank_read_no_origin_no_end(self):
         """GenBank.read(...) error on malformed file."""
-        with open(os.path.join("GenBank", "no_origin_no_end.gb")) as handle:
+        path = os.path.join("GenBank", "no_origin_no_end.gb")
+        with open(path) as handle:
             self.assertRaises(ValueError, GenBank.read, handle)
 
     # Evil hack with 000 to manipulate sort order to ensure this is tested
     # first (otherwise something silences the warning)
     def test_000_genbank_bad_loc_wrap_warning(self):
         """Feature line wrapping warning."""
+        path = os.path.join("GenBank", "bad_loc_wrap.gb")
         with warnings.catch_warnings():
             warnings.simplefilter("error", BiopythonParserWarning)
-            with open(os.path.join("GenBank", "bad_loc_wrap.gb")) as handle:
-                # self.assertRaises(BiopythonParserWarning, GenBank.read, handle)
-                try:
-                    record = GenBank.read(handle)
-                except BiopythonParserWarning as e:
-                    self.assertEqual(str(e),
-                                     "Non-standard feature line wrapping (didn't break on comma)?")
-                else:
-                    self.assertTrue(False, "Expected specified BiopythonParserWarning here.")
+            with open(path) as handle:
+                with self.assertRaises(BiopythonParserWarning) as cm:
+                    GenBank.read(handle)
+                self.assertEqual("Non-standard feature line wrapping (didn't break on comma)?", str(cm.exception))
```

Comment:
What does variable name ``cm`` mean? If you want short and sweet, how about ``w`` for warning?

Reply:
cm = context manager